scripts/ci/test_ids/test_id_checking_lib.py

```diff
@@ -0,0 +1,148 @@
+#!/usr/bin/env python3
+
+# SPDX-FileCopyrightText: 2025 3mdeb <contact@3mdeb.com>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import ast
+import difflib
+import os
+import re
+import subprocess
+from pathlib import Path
+
+from robot.model import SuiteVisitor
+from robot.running import TestSuiteBuilder
+
+
+class TestCasesFinder(SuiteVisitor):
+    def __init__(self):
+        self.tests = []
+
+    def visit_test(self, test):
+        self.tests.append(test)
+
+
+class OsConfig:
+    def __init__(self):
+        """Parse the os-config file contents into a dictionary"""
+        self.os_ids = {}
+        self.friendly_names = {}
+        this_file = os.path.dirname(os.path.realpath(__file__))
+        os_config_path = os.path.join(
+            #               ci scripts osfv
+            this_file,
+            "..",
+            "..",
+            "..",
+            "os-config",
+            "environment-test-ids.py",
+        )
+        self._parse_os_config(os_config_path)
+
+    def _parse_os_config(self, path):
+        src = Path(path).read_text()
+        tree = ast.parse(src, filename=path)
+        vars = {}
+
+        def _parse_ast_node(node):
+            if isinstance(node, ast.Constant):  # literal
+                return node.value
+            if isinstance(node, ast.Name):  # reference
+                try:
+                    return vars[node.id]
+                except KeyError as exc:
+                    raise ValueError(f"Unknown identifier {node.id!r}") from exc
+            if isinstance(node, (ast.List, ast.Tuple, ast.Set)):  # containers
+                ctor = {ast.List: list, ast.Tuple: tuple, ast.Set: set}[type(node)]
+                return ctor(_parse_ast_node(elt) for elt in node.elts)
+            if isinstance(node, ast.Dict):
+                return {
+                    _parse_ast_node(k): _parse_ast_node(v)
+                    for k, v in zip(node.keys, node.values)
+                }
+            raise ValueError(f"Unsupported expression: {ast.dump(node)}")
+
+        for node in tree.body:
+            if (
+                isinstance(node, ast.Assign)
+                and len(node.targets) == 1
+                and isinstance(node.targets[0], ast.Name)
+            ):
+                name = node.targets[0].id
+                vars[name] = _parse_ast_node(node.value)
+
+        for name, value in vars.items():
+            if name == "ENV_ID_OS_FRIENDLY_NAMES":
+                self.friendly_names = value
+            else:
```

Comment:
```suggestion
            elif not isinstance(value, dict):
```

Otherwise `os_id` in line 84 is unhashable and can't be compared against `friendly_names` key.